File: app/src/main/java/com/kickstart/application/di/component/ApplicationComponent.java
package com.kickstart.application.di.component;


import com.kickstart.application.AndroidApplication;
import com.kickstart.application.di.module.ApplicationModule;
import com.kickstart.application.di.module.BinderModule;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.support.AndroidSupportInjectionModule;

/**
 * Lifetime is throughout the application lifecycle.
 */
@Singleton
@Component(modules = {
        AndroidSupportInjectionModule.class,
        ApplicationModule.class,
        BinderModule.class
})
public interface ApplicationComponent {
    void inject(AndroidApplication androidApplication);
}

File: app/src/main/java/com/kickstart/railway/di/component/ApplicationComponent.kt
package com.kickstart.railway.di.component


import com.kickstart.railway.AndroidApplication
import com.kickstart.railway.di.module.ApplicationModule
import com.kickstart.railway.di.module.BinderModule

import javax.inject.Singleton

import dagger.Component
import dagger.android.support.AndroidSupportInjectionModule

/**
 * Lifetime is throughout the application lifecycle.
 */
@Singleton
@Component(modules = arrayOf(AndroidSupportInjectionModule::class, ApplicationModule::class, BinderModule::class))
interface ApplicationComponent {
    fun inject(androidApplication: AndroidApplication)
}
